proxy/http2/Http2ConnectionState.cc

```diff
@@ -1497,6 +1517,12 @@ Http2ConnectionState::send_a_data_frame(Http2Stream *stream, size_t &payload_len
     return Http2SendDataFrameResult::ERROR;
   }
 
+  SCOPED_MUTEX_LOCK(lock, this->ua_session->mutex, this_ethread());
```

Comment:
this lock it's being re-added here after being removed as part of #6337, do we really need it?

Reply:
Nice catch. This is an unnecessary lock, I think.

Reply:
If there's any path calling `send_a_data_frame` without acquiring the mutex, we should fix the caller.

Reply:
Since I created the PR (@masaori335 being the author of course) I updated the PR by removing the scoped mutex.

Reply:
Thanks!